components/date-time/time.js

```diff
@@ -93,6 +105,48 @@ class TimePicker extends Component {
 		onChange( formattedDate );
 	}
 
+	updateDay() {
+		const { onChange } = this.props;
+		const { day, date } = this.state;
+		const value = parseInt( day, 10 );
+		if ( ! isInteger( value ) || value < 1 || value > 31 ) {
+			this.syncState( this.props );
+			return;
+		}
+		const newDate = date.clone().date( value );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+		onChange( formattedDate );
+	}
+
+	updateMonth() {
+		const { onChange } = this.props;
+		const { month, date } = this.state;
+		const value = parseInt( month, 10 );
+		if ( ! isInteger( value ) || value < 1 || value > 12 ) {
+			this.syncState( this.props );
+			return;
+		}
+		const newDate = date.clone().month( value - 1 );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
+		onChange( formattedDate );
+	}
+
+	updateYear() {
+		const { onChange } = this.props;
+		const { year, date } = this.state;
+		const value = parseInt( year, 10 );
+		if ( ! isInteger( value ) || value < 1970 || value > 9999 ) {
+			this.syncState( this.props );
+			return;
+		}
+		const newDate = date.clone().year( value );
+		this.setState( { date: newDate } );
+		const formattedDate = newDate.format( TIMEZONELESS_FORMAT );
```

Comment:
Setting these as constants makes the intent clearer because they get useful names, but this variable is assigned the line before it's used, once.

It might be nicer to just comment anything you want to add clarification to and use the result of things like `newDate.format( TIMEZONELESS_FORMAT )` inline, eg:

```js
this.setState( { date: newDate } );
onChange( newDate.format( TIMEZONELESS_FORMAT ) )
```

Just a thought 🤷‍♂️

Reply:
True. Let me have a look really quickly. I've always found the classic JS coding style to be very heavy on assigning everything as a variable though.